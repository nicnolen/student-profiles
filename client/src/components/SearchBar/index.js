import React, { useState, useEffect } from 'react';
import UseFetch from '../UseFetch/index';
import StudentProfile from '../StudentProfile/index';

const SearchBar = () => {
  const [students] = UseFetch('https://api.hatchways.io/assessment/students');
  const [input, setInput] = useState('');
  const [tagSearch, setTagSearch] = useState([]);
  const [filterStudents, setFilterStudents] = useState([]);

  useEffect(() => {
    const filteredStudentsByNameAndTag = students.filter(student => {
      const firstName = student.firstName.toLowerCase();
      const lastName = student.lastName.toLowerCase();
      const fullName = firstName + ' ' + lastName;

      if (input === '' && tagSearch === '') {
        return student;
      } else if (fullName.includes(input.toLowerCase())) {
        if (tagSearch === '') {
          return student;
        } else if (
          student.tags.filter(tag => tag.toLowerCase().includes(tagSearch))
            .length >= 1
        ) {
          return student;
        }
      }
    });

    setFilterStudents(filteredStudentsByNameAndTag);
  }, [students, input, tagSearch]);

  return (
    <>
      <div>
        <input
          type="text"
          placeholder="Search by name"
          onChange={event => {
            setInput(event.target.value);
          }}
          className="nameSearchInput"
        />
        <input
          type="text"
          placeholder="Search by tag"
          onChange={event => {
            setTagSearch(event.target.value);
          }}
          className="tagSearchInput"
        />
      </div>
      <div>
        {filterStudents.map(student => (
          <StudentProfile key={student.id} student={student} />
        ))}
      </div>
    </>
  );
};

export default SearchBar;
