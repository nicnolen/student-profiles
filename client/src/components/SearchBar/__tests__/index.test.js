// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import SearchBar from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('SearchBar component', () => {
  // verify the component is rendering properly
  it('SearchBar component renders', () => {
    render(<SearchBar />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<SearchBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});