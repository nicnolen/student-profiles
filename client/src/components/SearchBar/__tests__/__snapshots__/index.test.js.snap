// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    <input
      class="nameSearchInput"
      placeholder="Search by name"
      type="text"
    />
    <input
      class="tagSearchInput"
      placeholder="Search by tag"
      type="text"
    />
  </div>
  <div />
</DocumentFragment>
`;
